src/System.Security.Cryptography.Pkcs/tests/SignedCms/SignerInfoTests.netcoreapp.cs

```diff
@@ -0,0 +1,270 @@
+        {
+            var indices = new HashSet<int>();
+            foreach (CryptographicAttributeObject attribute in attributes)
+            {
+                foreach (AsnEncodedData attributeValue in attribute.Values)
+                {
+                    int idx = FindAsnEncodedData(expectedAttributes, attributeValue);
+                    Assert.NotEqual(-1, idx);
+                    indices.Add(idx);
+                }
+            }
+
+            Assert.Equal(expectedAttributes.Count, indices.Count);
+        }
+
+        private static int FindAsnEncodedData(List<AsnEncodedData> array, AsnEncodedData data)
+        {
+            for (int i = 0; i < array.Count; i++)
+            {
+                if (AsnEncodingDataEqual(array[i], data))
+                {
+                    return i;
+                }
+            }
+
+            return -1;
+        }
+
+        private static List<AsnEncodedData> GetAllAsnEncodedDataFromAttributes(CryptographicAttributeObjectCollection attributes)
+        {
+            var ret = new List<AsnEncodedData>();
+            foreach (CryptographicAttributeObject attribute in attributes)
+            {
+                foreach (AsnEncodedData attributeValue in attribute.Values)
+                {
+                    ret.Add(attributeValue);
+                }
+            }
+
+            return ret;
+        }
+
+        private static bool AsnEncodingDataEqual(AsnEncodedData a, AsnEncodedData b)
+        {
+            return a.Oid.Value == b.Oid.Value && a.RawData.SequenceEqual(b.RawData);
+        }
+
+        private static SignedCms ReReadSignedCms(SignedCms cms)
+        {
+            byte[] bytes = cms.Encode();
+
+            cms = new SignedCms();
+            cms.Decode(bytes);
+            return cms;
+        }
+
+        private static AsnEncodedData CreateTimestampToken(byte serial)
+        {
+            Oid tokenOid = new Oid(TokenAttributeOid, TokenAttributeOid);
+
+            Oid policyId = new Oid("0.0", "0.0");
+            Oid hashAlgorithmId = new Oid(Oids.Sha256);
+
+            var tokenInfo = new Rfc3161TimestampTokenInfo(
+                policyId,
+                hashAlgorithmId,
+                new byte[256 / 8],
+                new byte[] { (byte)serial },
+                DateTimeOffset.UtcNow);
+
+            return new AsnEncodedData(tokenOid, tokenInfo.Encode());
+        }
+
+        private static byte Read8BitSerial(ReadOnlySpan<byte> serial)
+        {
+            // serial is big endian integer but we expect value in range 0-255
+            byte x = 0;
+            foreach (byte b in serial)
+            {
+                Assert.Equal(0, x);
+                x = b;
+            }
+
+            return x;
+        }
+
+        private static void CheckSignerHasUnsignedAttribute(AsnEncodedData actual, AsnEncodedData expected)
```

Comment:
The name of this method, the contents of this method, and the arguments of this method don't feel like they match up.

Reply:
true, forgot to rename when converting tests from CryptographicObjectAttribute version